src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
-
-    glTexImage2D(
-        GL_TEXTURE_2D,
-        0,
-        GL_RGBA32F,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        0,
-        GL_RGBA,
-        GL_FLOAT,
-        nullptr);
-
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-
-    glBindTexture(GL_TEXTURE_2D_MULTISAMPLE, _mainPositionTexture);
-
-    glTexImage2DMultisample(
-        GL_TEXTURE_2D_MULTISAMPLE,
-        _nAaSamples,
-        GL_RGBA32F,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        true);
-
-    glBindTexture(GL_TEXTURE_2D_MULTISAMPLE, _mainNormalTexture);
-
-    glTexImage2DMultisample(
-        GL_TEXTURE_2D_MULTISAMPLE,
-        _nAaSamples,
-        GL_RGBA32F,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        true);
-
-    glBindTexture(GL_TEXTURE_2D_MULTISAMPLE, _mainDepthTexture);
-    glTexImage2DMultisample(
-        GL_TEXTURE_2D_MULTISAMPLE,
-        _nAaSamples,
-        GL_DEPTH_COMPONENT32F,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        true
-    );
-
-    glBindTexture(GL_TEXTURE_2D, _exitColorTexture);
-    glTexImage2D(
-        GL_TEXTURE_2D,
-        0,
-        GL_RGBA16,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        0,
-        GL_RGBA,
-        GL_UNSIGNED_SHORT,
-        nullptr
-    );
-
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-
-    glBindTexture(GL_TEXTURE_2D, _exitDepthTexture);
-
-    glTexImage2D(
-        GL_TEXTURE_2D,
-        0,
-        GL_DEPTH_COMPONENT32F,
-        GLsizei(_resolution.x),
-        GLsizei(_resolution.y),
-        0,
-        GL_DEPTH_COMPONENT,
-        GL_FLOAT,
-        nullptr
-    );
-
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-
-    _dirtyResolution = false;
-}
-
-void FramebufferRenderer::updateRaycastData() {
-    _raycastData.clear();
-    _exitPrograms.clear();
-    _raycastPrograms.clear();
-    _insideRaycastPrograms.clear();
-
-    const std::vector<VolumeRaycaster*>& raycasters =
-        OsEng.renderEngine().raycasterManager().raycasters();
-    int nextId = 0;
-    for (auto& raycaster : raycasters) {
-        RaycastData data;
-        data.id = nextId++;
-        data.namespaceName = "HELPER";
-
-        std::string vsPath = raycaster->getBoundsVsPath();
-        std::string fsPath = raycaster->getBoundsFsPath();
+    void FramebufferRenderer::updateDeferredcastData() {
+        _deferredcastData.clear();
+        _deferredcastPrograms.clear();
+
+        const std::vector<Deferredcaster*>& deferredcasters =
+            OsEng.renderEngine().deferredcasterManager().deferredcasters();
+        int nextId = 0;
+        for (auto& caster : deferredcasters) {
```

Comment:
No `auto`;  use the correct type instead

Reply:
Done.